Add metadata to templates

francescaleung opened the pull request:
I have feelings about this; I'm not 100% sure how I want to handle images because they're currently statically imported & I don't know if we want to link to github on our docs/web app. Currently mucking around with the docs site to see what we're able to do there.

Resolves ASM-1029. Some work here for ASM-1024 but there'll probably be a better pass of that later.

calufornia commented on deployment_pipeline_dashboard/.metadata/metadata.json:
```diff
@@ -0,0 +1,27 @@
+    }
+  ],
+  "components": ["Button", "Divider", "Heading", "Link", "Stack", "Table"],
+  "tags": ["type:task", "type:view", "task:ts", "thirdparty:GitHub"],
```
I noticed that some thirdparty tags are properly capitalized (e.g. here) but others are lowercase (e.g. `aws`). Can we standardize on one (preferably the former)?

calufornia commented on customer_onboarding_dashboard/.metadata/preview.png:
This template needs corresponding metadata

francescaleung replied:
Deliberately leaving this out since it's not showcased on the docs site

calufornia commented on scripts/generate_metadata.py:
```diff
@@ -0,0 +1,53 @@
+    all_metadata = {}
+    for path in os.listdir(root):
+        metadata_folder = os.path.join(root, path, AIRPLANE_METADATA_FOLDER)
+        if not os.path.isdir(os.path.join(root, path)):
+            continue
+        if not os.path.isdir(metadata_folder):
+            continue
```
```suggestion
        if not os.path.isdir(metadata_folder):
            continue
```

Do we need the first check?

calufornia reviewed (commented):
Can we add a README on how to regenerate the global metadata.json file? Can we also add a check to CI to ensure that this file is up to date? These can be in a separate PR if you want

francescaleung commented:
> Can we add a README on how to regenerate the global metadata.json file? Can we also add a check to CI to ensure that this file is up to date? These can be in a separate PR if you want

Added this to CONTRIBUTING.md instead, since I think it's more relevant there. Also added a CONTRIBUTING.md file.

calufornia commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,33 @@
+directory, which should include a `metadata.json` with the metadata for the template and a `preview.png` that includes
+the preview screenshot for the template. See existing templates for examples.
+
+Your metadata file should have _at minmimum_ the following keys:
```
```suggestion
Your metadata file should have _at minimum_ the following keys:
```

calufornia commented on .github/CONTRIBUTING.md:
````diff
@@ -0,0 +1,33 @@
+a corresponding `requirements.txt` file in `scripts` that you'll need to install.
+
+```
+$ python -m venv venv       # optional but recommended
+$ source venv/bin/activate  # only necessary if you're using a venv
````
```suggestion
$ python -m venv .venv       # optional but recommended
$ source .venv/bin/activate  # only necessary if you're using a venv
```
`.venv` is a little more standard

calufornia commented on .github/workflows/generate.yaml:
```diff
@@ -0,0 +1,27 @@
+          GIT_DIFF=$(git diff)
+
+          if [[ $GIT_STATUS ]]; then
+            echo "Found format diffs:"
```
```suggestion
            echo "Found metadata diffs:"
```

calufornia commented on scripts/generate_metadata.py:
```diff
@@ -0,0 +1,136 @@
+        cat, value = tag.split(":")
+        if cat == "type":
+            num_type_tags += 1
+        if value[0] not in string.ascii_uppercase:
```
`string.ascii_uppercase`: TIL

calufornia reviewed (approved):
I feel like we shouldn't clone the `.metadata` folder when initializing the template. Can you add logic to the CLI to filter that folder out?

calufornia commented:
Looks great!
